snapshot/src/main/java/io/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -364,10 +390,28 @@ PersistedSnapshot newSnapshot(final FileBasedSnapshotMetadata metadata, final Pa
       return currentPersistedSnapshot;
     }
 
+    // it's important to persist the checksum file only after the move is finished, since we use it
+    // as a marker file to guarantee the move was complete and not partial
     final var destination = buildSnapshotDirectory(metadata);
     moveToSnapshotDirectory(directory, destination);
 
-    final var newPersistedSnapshot = new FileBasedSnapshot(destination, metadata);
+    final var checksumPath = buildSnapshotsChecksumPath(metadata);
+    final long actualChecksum;
+    try {
+      // computing the checksum on the final destination also lets us detect any failures during the
+      // copy/move that could occur
+      actualChecksum = SnapshotChecksum.calculate(destination);
```

Comment:
It was intended, but as I wrote in the description, I would like to challenge it. As you mentioned, it can be expensive if we have a large snapshot. At the same time, it's more or less the only way to verify that the snapshot was moved and kept its integrity. If we think this is safe to do and only calculate it after the move, we could do so as well, but I was unsure about it.

Reply:
We can discuss it later also, as it is not a breaking change. One thing that I discussed with Nico was to only verify the number of files after moving. The reasoning was - The checksum is calculated in the same context as the move, so there is nothing that can happen in between that can corrupt the snapshot. Just to make sure that move is complete, we can compare the number of files.